arch/arm/src/stm32wl5/stm32wl5_ipcc.c

```diff
@@ -0,0 +1,801 @@
+      .txlen = IPCC_CHAN2_TX_SIZE - sizeof(unsigned)
+    }
+#endif
+
+#if IPCC_CHAN3
+    ,
+    {
+      .rxmem = (char *)(IPCC_CHAN3_START),
+      .rxlen = IPCC_CHAN3_RX_SIZE - sizeof(unsigned),
+      .txmem = (char *)(IPCC_CHAN3_START + IPCC_CHAN3_RX_SIZE),
+      .txlen = IPCC_CHAN3_TX_SIZE - sizeof(unsigned)
+    }
+#endif
+
+#if IPCC_CHAN4
+    ,
+    {
+      .rxmem = (char *)(IPCC_CHAN4_START),
+      .rxlen = IPCC_CHAN4_RX_SIZE - sizeof(unsigned),
+      .txmem = (char *)(IPCC_CHAN4_START + IPCC_CHAN4_RX_SIZE),
+      .txlen = IPCC_CHAN4_TX_SIZE - sizeof(unsigned)
+    }
+#endif
+
+#if IPCC_CHAN5
+    ,
+    {
+      .rxmem = (char *)(IPCC_CHAN5_START),
+      .rxlen = IPCC_CHAN5_RX_SIZE - sizeof(unsigned),
+      .txmem = (char *)(IPCC_CHAN5_START + IPCC_CHAN5_RX_SIZE),
+      .txlen = IPCC_CHAN5_TX_SIZE - sizeof(unsigned)
+    }
+#endif
+
+#if IPCC_CHAN6
+    ,
+    {
+      .rxmem = (char *)(IPCC_CHAN6_START),
+      .rxlen = IPCC_CHAN6_RX_SIZE - sizeof(unsigned),
+      .txmem = (char *)(IPCC_CHAN6_START + IPCC_CHAN6_RX_SIZE),
+      .txlen = IPCC_CHAN6_TX_SIZE - sizeof(unsigned)
+    }
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: stm32wl5_ipcc_tx_isr
+ *
+ * Description:
+ *   IPCC TX interrupt service routine. This interrupt is called when
+ *   second CPU read all data from IPCC TX memory and we are free to
+ *   write new data to that memory.
+ *
+ *   For buffered IPCC, we will immediately write data from buffer to
+ *   IPCC memory and notify second CPU. Also blocked writer will be
+ *   notified that there is free space on tx buffer.
+ *
+ *   For unbuffered IPCC, we won't copy anything but only notify blocked
+ *   writers that IPCC TX memory is free.
+ *
+ * Input Parameters:
+ *   irq - Number of IRQ that generated interrupt
+ *   context - Interrupt register state save info (architecture-specific)
+ *   arg - user data (not used by us)
+ *
+ * Returned Value:
+ *   Always OK
+ *
+ * Assumptions/Limitations:
+ *
+ ****************************************************************************/
+
+static int stm32wl5_ipcc_tx_isr(int irq, void *context, void *arg)
+{
+  int chan;
+  size_t nwritten;
+  uint32_t mr;
+  uint32_t sr;
+  uint32_t status;
+  struct stm32wl5_ipcc_s *priv;
+  struct stm32wl5_ipcc_chan_mem_s *txmem;
+
+  (void)context;
+  (void)arg;
+  (void)irq;
+
+  /* Disable TX interrupt, we don't want two "threads" to concurently
+   * modify circbuffers as we don't use any guard to them
+   */
+
+  up_disable_irq(STM32WL5_IRQ_IPCC_C1_TX_IT);
+
+  mr = getreg32(STM32WL5_IPCC_C1MR) >> STM32WL5_IPCC_TX_SHIFT;
+  sr = getreg32(STM32WL5_IPCC_C1TOC2SR);
+
+  /* Consider only channels that have tx memory free and are unmasked */
+
+  status = sr | mr;
```

Comment:
Looks like you are expecting `AND` logic, but the written is `OR`.

Reply:
Actually I did mean OR here.

If status is ```true``` then transmit channel is either masked or memory is not ready to be written. sr == 1, means "Channel occupied data can be read by second cpu" - so we cannot write to it. mr == 1, means channel is masked. So if any of these values are 1, then we cannot write to the memory.